### ISSUE TITLE
Edit Content: Expose the `editableAsText` even if the metadata file doesn't have it yet

### ISSUE THREAD

**Issue by jcastro-dotcms** [dotCMS/core#27393]
### Parent Issue

#26910

### Problem Statement

The new `editableAsText` property is available to the REST endpoints as long as the File Asset in dotCMS has been either re-saved or re-indexed before accessing it. This may represent a performance issue in customers with hundreds of thousands of files, and we need the property to be available right away.

### Steps to Reproduce

1. Start up dotCMS version 23.10 or lower. Just make sure that it was created before December 18th, 2023.
2. Use the Full Starter to have files you can test with. Let dotCMS start up and make sure it looks ok.
3. Shut it down, and re-start it using the latest master image.
5. Open up the `/application/vtl/activities/activities-detail.vtl` file.
6. You'll see that the File Asset will NOT display the code editor for you to edit its contents directly, which is wrong. **IMPORTANT: DO NOT click `Save` or `Publish` as this will make the issue non-reproducible.**

### Acceptance Criteria

dotCMS must be able to figure out whether the selected file can be editable directly in the back-end or not whitout relying exclusively on recently indexed metadata. If it can, then its contents must be displayed in a Code Editor field.

### dotCMS Version

Latest master.

### Proposed Objective

Core Features

### Proposed Priority

Priority 2 - Important

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

https://dotcms.slack.com/archives/CQQMX585N/p1705592078210759

**Comment by wezell** [comment]
Still seeing this on non-published files:

https://github.com/dotCMS/core/assets/934364/39dac69b-d83c-4d3f-962c-2b0d4ee52155

**Comment by jcastro-dotcms** [comment]
@wezell the reason why the `metaData` attribute is not present is because the actual binary files for those specific Inodes are missing. If you go to https://authoring.dotcms.com/dotAdmin/#/c/content and do the following:

* Select `File Asset` in the **Type** dropdown.
* Click the `Advanced` button, and select `Unpublished` in the **Show** dropdown.

You'll see that all four entries in there have a `NOT_FOUND` value in the **File Asset** column:

<img width="1375" alt="Screenshot 2024-02-08 at 4 18 07 PM" src="https://github.com/dotCMS/core/assets/8779585/c020aacf-7db5-432a-82a3-4c04da20e901">

Yesterday, I worked with Manuel Miranda to check for the existence of those binary files in the `/assets/` folder, but he SSH'ed into the incorrect server. Today, we were able to confirm that those binary files are gone indeed. That's also the reason why downloading the `starter.zip` file via the **Settings > Maintenance** portlet didn't include them either, and not because of an error with the starter generation process.

I'll be adding some code changes to report the missing binary situation properly along with the improvements we discussed via Slack.
